File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n=height.size();
        int ans=0;

        int i=0;
        int j=n-1;
        
        while(i<=j){
            int len=j-i;
            int area=len*min(height[i],height[j]);
            ans=max(ans,area);
            if (height[i]<=height[j]) i++;
            else j--;
        }
        return ans;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n=height.length;
        int ans=Integer.MIN_VALUE;

        int l=0,r=n-1;
        while(l<r){
            int hei=Math.min(height[l],height[r]);
            int len=r-l;
            ans=Math.max(ans,hei*len);
            if (height[l]<=height[r]) l++;
            else r--;
        }
        return ans;
    }
}